src/Communities/algorithms/pagerank.jl

```diff
@@ -1,74 +1,221 @@
+            adj_matrix[row_index, :] .= 1
+            if loop == 0
+                adj_matrix[row_index, row_index] = 0
             end
-            PR[i] = (1 - algo.d) + algo.d * s
         end
+    end
+    
+    # normalize rows to sum to 1 and transpose
+    P = adj_matrix ./ sum(adj_matrix, dims=2)
+    P = copy(SparseArrays.transpose(P))
+
+    f_map = Dict("classical" => _classical_pagerank, "iterative" => _iterative_regularization_pagerank, "adaptive" => _adaptive_regularization_pagerank)
+
+    pagerank_function = f_map[algo.method_name]
+    d = algo.d
+    tol = algo.tol
+    max_iter = algo.max_iter
+
+    return pagerank_function(P, d, tol, max_iter)
+end  
+
+"""
+    _classical_pagerank(P::SparseMatrixCSC, d::Float64, tol::Float64, max_iter::Int64)::Vector{Float64}
+
+Helper function for compute(Pagerank, g), computes the PageRank values of the nodes in graph `g` using the classical PageRank algorithm.
 
-        # Check for convergence
-        if maximum(abs.(PR - old_PR)) < algo.tol
-            break
+# Arguments
+
+- `P::SparseMatrixCSC`: The Markov Chain associated with `g`.
+- `d::Float64`: The damping factor.
+- `tol::Float64`: The tolerance for convergence.
+- `max_iter::Int64`: The maximum number of iterations.
+
+# Returns
+
+- A vector of `Float64` where each entry represents the PageRank value of the corresponding node in the graph.
+"""
+function _classical_pagerank(
+    P::SparseMatrixCSC, 
+    d::Float64, 
+    tol::Float64, 
+    max_iter::Int64
+)::Vector{Float64}
+
+    # initialize
+    N = size(P, 1)
+
+    pr = LinearAlgebra.ones(N) / N
+
+    for _ in 1:max_iter
+        prev_pr = copy(pr)
+        # power method for principle eigenvector of modified P
+        pr = d .* P * pr .+ (1 - d) / N
+        
+        if LinearAlgebra.norm(pr - prev_pr, 1) < tol
+            return pr
         end
-        old_PR = copy(PR)
     end
 
-    # Normalize
-    PR ./= sum(PR)
+    println("Warning: max_iter reached")
+    return pr
+end
+
+"""
+    _iterative_pagerank(P::SparseMatrixCSC, d::Float64, tol::Float64, max_iter::Int64)::Vector{Float64}
+
+Helper function for compute(Pagerank, g), computes the PageRank values of the nodes in graph `g` using the iterative regularization PageRank algorithm.
 
-    return PR
+# Arguments
+
+- `P::SparseMatrixCSC`: The Markov Chain associated with `g`.
+- `d::Float64`: The initial damping factor, from which the regularization factor is computed. 0 < d < .5
+- `tol::Float64`: The tolerance for convergence.
+- `max_iter::Int64`: The maximum number of iterations.
+
+# Returns
+
+- A vector of `Float64` where each entry represents the PageRank value of the corresponding node in the graph.
+"""
+function _iterative_regularization_pagerank(
+    P::SparseMatrixCSC, 
+    d::Float64, 
+    tol::Float64, 
+    max_iter::Int64
+)::Vector{Float64}
+
+    # initialize
+    N = size(P, 1)
+
+    pr = LinearAlgebra.ones(N) / N
+    # compute p for regularization factor
```

Comment:
From now on let us try and use complete sentences for any comment we add to code. This includes capitalization and use of punctuation.